flexget/entry.py

```diff
@@ -87,7 +87,12 @@ def __init__(self, *args, **kwargs):
         # Make sure constructor does not escape our __setitem__ enforcement
         self.update(*args, **kwargs)
 
-    def trace(self, message: str, operation: Optional[str] = None, plugin: Optional[str] = None):
+    def trace(
+        self,
+        message: Optional[str],
```

Comment:
Looks like the optional bit got moved to message here instead of the other 2 arguments.

Reply:
At first I made the parameters' annotation follow the `Optional[<type>] = None` approach, but then I noticed that you were not using the [`no_implicit_optional`](https://mypy.readthedocs.io/en/stable/config_file.html#confval-no_implicit_optional) config option (it's `False` by default). When it's not enabled the following 2 annotations for arguments are equivalent:
1. `var: Optional[<type>] = None`
2. `var: <type> = None`

The latter seems more readable, so while converting my initial annotations to the 2nd form I changed some more (like the one above) for consistency. `message` was annotated as `Optional[str]`, as it's not initialised with `None`, but can receive `None` values (e.g. `accept()`, on line 170)

Reply:
Aha. I thought there must be some reason. I haven't used mypy much, and didn't know about this. In that case I'm good with it.